Index: servoPrepare.c
===================================================================
--- servoPrepare.c	(revision 1212)
+++ servoPrepare.c	(working copy)
@@ -39,8 +39,14 @@
 		udb_pwIn[i] = udb_pwTrim[i] = ((i == THROTTLE_INPUT_CHANNEL) ? 0 : 3000) ;	
 #endif
 
+#if (FIXED_TRIMPOINT == 1)
 	for (i=0; i <= NUM_OUTPUTS; i++)
+		udb_pwOut[i] = ((i == THROTTLE_OUTPUT_CHANNEL) ? THROTTLE_TRIMPOINT : CHANNEL_TRIMPOINT) ;
+#else
+	for (i=0; i <= NUM_OUTPUTS; i++)
 		udb_pwOut[i] = ((i == THROTTLE_OUTPUT_CHANNEL) ? 0 : 3000) ;
+#endif
+
 	
 #if (NORADIO == 1)
 	udb_pwIn[MODE_SWITCH_INPUT_CHANNEL] = udb_pwTrim[MODE_SWITCH_INPUT_CHANNEL] = 4000 ;
