Index: libUDB/libUDB.c
===================================================================
--- libUDB/libUDB.c	(revision 1206)
+++ libUDB/libUDB.c	(working copy)
@@ -190,10 +190,11 @@
 
 void udb_servo_record_trims(void)
 {
+#if (FIXED_TRIMPOINT != 1)	// Do not alter trims from preset when they are fixed
 	int i;
 	for (i=0; i <= NUM_INPUTS; i++)
 		udb_pwTrim[i] = udb_pwIn[i] ;
-	
+#endif	
 	return ;
 }
 
Index: libUDB/radioIn_udb4.c
===================================================================
--- libUDB/radioIn_udb4.c	(revision 1206)
+++ libUDB/radioIn_udb4.c	(working copy)
@@ -49,8 +49,16 @@
 void udb_init_capture(void)
 {
 	int i;
+	
 	for (i=0; i <= NUM_INPUTS; i++)
-		udb_pwIn[i] = udb_pwTrim[i] = 0 ;
+#if (FIXED_TRIMPOINT == 1)
+		if(i == THROTTLE_OUTPUT_CHANNEL)
+			udb_pwTrim[i] = udb_pwIn[i] = THROTTLE_TRIMPOINT;
+		else
+			udb_pwTrim[i] = udb_pwIn[i] = CHANNEL_TRIMPOINT;			
+#else
+		udb_pwTrim[i] = udb_pwIn[i] = 0 ;
+#endif
 	
 	TMR2 = 0 ; 				// initialize timer
 	T2CONbits.TCKPS = 1 ;	// prescaler = 8 option
Index: MatrixPilot/options.h
===================================================================
--- MatrixPilot/options.h	(revision 1206)
+++ MatrixPilot/options.h	(working copy)
@@ -308,6 +308,19 @@
 
 
 ////////////////////////////////////////////////////////////////////////////////
+// FIXED TRIMPOINT
+// Instead of sampling the radio inputs at startup, use a fixed value
+// This can be used if you would prefer a static trimpoint between reboots
+// Set FIXED_TRIMPOINT to 1 to use this feature
+// Channel trimpoint is for all channel excluding throttle.  Units in microseconds
+// Throttle trimpoint is only for the throttle channel.
+// NOTE: UDB trimpoints are in 0.5us units.
+#define FIXED_TRIMPOINT					0
+#define CHANNEL_TRIMPOINT				3040
+#define THROTTLE_TRIMPOINT				2250
+
+
+////////////////////////////////////////////////////////////////////////////////
 // Serial Output Format (Can be SERIAL_NONE, SERIAL_DEBUG, SERIAL_ARDUSTATION, SERIAL_UDB,
 // SERIAL_UDB_EXTRA,SERIAL_MAVLINK, SERIAL_CAM_TRACK, or SERIAL_OSD_REMZIBI)
 // This determines the format of the output sent out the spare serial port.
Index: MatrixPilot/servoPrepare.c
===================================================================
--- MatrixPilot/servoPrepare.c	(revision 1206)
+++ MatrixPilot/servoPrepare.c	(working copy)
@@ -33,8 +33,12 @@
 {
 	int i;
 	for (i=0; i <= NUM_INPUTS; i++)
-		udb_pwTrim[i] = udb_pwIn[i] = ((i == THROTTLE_INPUT_CHANNEL) ? 0 : 3000) ;
-	
+#if (FIXED_TRIMPOINT == 1)
+		udb_pwTrim[i] = udb_pwIn[i] = ((i == THROTTLE_INPUT_CHANNEL) ? THROTTLE_TRIMPOINT : CHANNEL_TRIMPOINT ) ;
+#else
+		udb_pwIn[i] = udb_pwTrim[i] = ((i == THROTTLE_INPUT_CHANNEL) ? 0 : 3000) ;	
+#endif
+
 	for (i=0; i <= NUM_OUTPUTS; i++)
 		udb_pwOut[i] = ((i == THROTTLE_OUTPUT_CHANNEL) ? 0 : 3000) ;
 	
