Index: libDCM/libDCM.c
===================================================================
--- libDCM/libDCM.c	(revision 1147)
+++ libDCM/libDCM.c	(working copy)
@@ -38,10 +38,12 @@
 									0x0D, 0x0E,		//S6
 									0x0F, 0x10,		//S7
 									0x11, 0x12,		//S8
-									0x13, 0x14		//checksum
+									0x13, 0x14,		//S9
+									0x15, 0x16,		//S10
+									0x17, 0x18		//checksum
 									};
 
-#define HILSIM_NUM_SERVOS 8
+#define HILSIM_NUM_SERVOS 10
 
 void send_HILSIM_outputs( void ) ;
 #endif
Index: Tools/XP_UDB_HILSIM/HILSIM.cpp
===================================================================
--- Tools/XP_UDB_HILSIM/HILSIM.cpp	(revision 1147)
+++ Tools/XP_UDB_HILSIM/HILSIM.cpp	(working copy)
@@ -48,15 +48,17 @@
 
 void	(* msg_parse) (unsigned char rxChar) = &msgDefault;
 
-#define SERVO_MSG_LENGTH 16
+#define SERVO_CHANNELS		10
 
+#define SERVO_MSG_LENGTH	SERVO_CHANNELS*2
+
 // Servo offsets are a variable so that they can be actively zeroed from the real received offsets
 // This offset zero will need to be a menu add on to the plugin.
 intbb  ServoOffsets[SERVO_CHANNELS];
 
-unsigned char SERVO_IN[SERVO_CHANNELS*2];
+unsigned char SERVO_IN[SERVO_MSG_LENGTH];
 
-unsigned char SERVO_IN_[SERVO_CHANNELS*2];
+unsigned char SERVO_IN_[SERVO_MSG_LENGTH];
 
 int rxCount = 0;
 
