Index: mavutil.py
===================================================================
--- mavutil.py	(revision 1260)
+++ mavutil.py	(working copy)
@@ -10,10 +10,10 @@
 from math import *
 from mavextra import *
 
-if os.getenv('MAVLINK10'):
-    import mavlinkv10 as mavlink
-else:
-    import mavlink
+#if os.getenv('MAVLINK10'):
+import mavlinkv10 as mavlink
+#else:
+#    import mavlink
 
 def evaluate_expression(expression, vars):
     '''evaluation an expression'''
